hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -156,6 +159,8 @@ public KeyValueHandler(ConfigurationSource config,
         DatanodeConfiguration.class).isChunkDataValidationCheck();
     chunkManager = ChunkManagerFactory.createChunkManager(config, blockManager,
         volSet);
+    checksumManager = new ContainerChecksumTreeManager(
+        config.getObject(DatanodeConfiguration.class));
```

Comment:
We need one checksum manager per datanode, and all services like scanner and block deletion need to use the same instance for coordination. For this reason it will probably need to get passed in from the [OzoneContainer](https://github.com/apache/ozone/blob/1737194a62bcc879eddbae9b7057a7e75f02d6c6/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java#L190). This is how it is currently done in the [block deletion patch](https://github.com/apache/ozone/pull/6875/files#r1688351623).

In later patches where the `ReconcileContainerCommandHandler` also needs this, it might get moved up to the [DatanodeStateMachine](https://github.com/apache/ozone/blob/8676000a6f6a1fe50b6d6764447826bf58a85548/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java#L182). For now, let's put it in the `OzoneContainer` to be consistent with the block delete change and we can move it again later if needed.

Reply:
Agreed. I added it as a part of the previous iteration of this change. I'll remove it here, we can add it in further patches when we connect these components.

Reply:
One problem I see is that `handleGetContainerMerkleTree` requires checksum manager. We need to either move this or add checksum manager to `KeyValueHandler` again it would lead to more number of file changes.